src/playlist-loader.js

```diff
@@ -268,11 +330,60 @@ export default class PlaylistLoader extends EventTarget {
         // only refresh the media playlist if no other activity is going on
         return;
       }
+      const media = this.media();
+
+      let uri = resolveUrl(this.master.uri, media.uri);
+
+      if (this.experimentalLLHLS && !media.endList) {
+        const query = [];
+
+        if (media.serverControl && media.serverControl.canBlockReload) {
+          const {preloadSegment} = media;
+          // next msn is a zero based value, length is not.
+          let nextMSN = media.mediaSequence + media.segments.length;
+
+          if (preloadSegment) {
+            const parts = preloadSegment.parts || [];
+            // _HLS_part is a zero based index
+            const nextPart = getKnownPartCount(media) - 1;
+
+            // if nextPart is > -1 and not equal to just the
+            // length of parts, then we know we had part preload hints
+            // and we need to add the _HLS_part= query
+            if (nextPart > -1 && nextPart !== (parts.length - 1)) {
+              // add existing parts to our preload hints
+              query.push(`_HLS_part=${nextPart}`);
+            }
+
+            // if we are requesting a nextPart or preload segment was added
+            // to our segment list. We are requesting a part of the preload segment
+            // or the full preload segment. Either way we need to go down by 1
+            // in nextMSN
+            if (nextPart > -1 || parts.length) {
+              nextMSN--;
+            }
+          }
+
+          // add _HLS_msn= infront of any _HLS_part query
+          query.unshift(`_HLS_msn=${nextMSN}`);
+        }
+
+        if (media.serverControl && media.serverControl.canSkipUntil) {
+          // add _HLS_skip= infront of all other queries.
+          query.unshift('_HLS_skip=' + (media.serverControl.canSkipDateranges ? 'v2' : 'YES'));
```

Comment:
In order to separate concerns, maybe we could have a separate function which can add LL-HLS params to a URI.

Reply:
Done